### PR TITLE
C++: Split MemberFunction.qll from Function.qll.

### PR DESCRIPTION
**Pull request by geoffw0**
I'm making some changes to Function.qll in another branch, and realized I wanted to split this large file in two first, moving some of the less core stuff out to improve navigability.  But this is the kind of change that's best split off into a PR of it's own, so here it is.

### CODE REVIEW

**Comment by geoffw0** [comment]
Fixed QLDoc.

File: cpp/ql/src/semmle/code/cpp/MemberFunction.qll [hunk 1]
```diff
@@ -1,4 +1,9 @@
-import cpp
+/**
+ * Provides classes for working with C++ constructors, destructors,
+ * and user-defined operators.
+ */
```
Comment (MathiasVP) [inline cpp/ql/src/semmle/code/cpp/MemberFunction.qll:4]: Shouldn't this also mention member functions somewhere? Like:

```codeql
/**
 * Provides classes for working with C++ member functions, constructors, destructors,
 * and user-defined operators.
 */
```

Comment (geoffw0) [reply cpp/ql/src/semmle/code/cpp/MemberFunction.qll]: Yes it clearly should - thank you.  Done.

**Comment by geoffw0** [comment]
Thanks for the speedy merge.

**Pull request merged**